packages/salesforcedx-vscode-apex/src/oas/generationStrategy/methodByMethodStrategy.ts

```diff
@@ -5,23 +5,99 @@
  * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
 
+import * as fs from 'fs';
+import { DocumentSymbol } from 'vscode';
+import * as yaml from 'yaml';
 import {
   ApexClassOASEligibleResponse,
   ApexClassOASGatherContextResponse,
+  ApexOASClassDetail,
+  ApexOASMethodDetail,
   PromptGenerationResult,
-  PromptGenerationStrategyBid
+  PromptGenerationStrategyBid,
+  OpenAPIDoc
 } from '../../openApiUtilities/schemas';
 import { IMPOSED_FACTOR, PROMPT_TOKEN_MAX_LIMIT, RESPONSE_TOKEN_MAX_LIMIT, SUM_TOKEN_MAX_LIMIT } from '.';
 import { GenerationStrategy } from './generationStrategy';
-
+import { prompts } from './prompts';
 export const METHOD_BY_METHOD_STRATEGY_NAME = 'MethodByMethod';
 export class MethodByMethodStrategy extends GenerationStrategy {
-  callLLMWithPrompts(): Promise<string[]> {
-    throw new Error('Method not implemented.');
+  async callLLMWithPrompts(): Promise<string[]> {
+    try {
+      const llmService = await this.getLLMServiceInterface();
+
+      // Filter valid prompts and map them to promises
+      const responsePromises = this.prompts.filter(p => p?.length > 0).map(prompt => llmService.callLLM(prompt));
+
+      // Execute all LLM calls in parallel and store responses
+      this.llmResponses = await Promise.all(responsePromises);
```

Comment:
@mingxuanzhangsfdx If one promise fails then they all do. Is that what you want. If you would like an opportunity to examine each of these, then have a look at Promise.allSettled